craft_providers/lxd/lxd_provider.py

```diff
@@ -120,14 +114,26 @@ def launched_environment(
 
         :raises LXDError: if instance cannot be configured and launched
         """
-        image_remote = configure_buildd_image_remote()
+        image = get_remote_image(build_base)
+        image.add_remote()
+
+        # we can't guarantee daily and devel images, so explicitly warn the user
+        if not image.is_stable:
+            warnings.warn(
```

Comment:
I thought `warnings.warn()` was meant more for devs, and not end-users

Reply:
I believe @sergiusens requested this message be emitted with the `warnings` module so that it cannot be silenced or filtered out.

Reply:
My thought process was that we were enabling the devel images early and its used would be something an application would enable, not a user through configuration. I would for example slightly change the API and add an allow_unstable=False param here and raise your warning as an exception.

Reply:
As discussed offline, I changed this to an exception and made the `is_stable` property implicit.